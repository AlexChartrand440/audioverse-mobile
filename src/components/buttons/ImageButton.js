import React from 'react'
import PropTypes from 'prop-types'
import { TouchableOpacity, Image } from 'react-native'

const ImageButton = ({ onPress, style, imageStyle, accessibilityLabel, ...props }) => (
  <TouchableOpacity
    onPress={onPress}
    style={style}
    accessible={true}
    accessibilityLabel={accessibilityLabel}
    accessibilityRole="button">
    <Image {...props} style={imageStyle} />
  </TouchableOpacity>
)

ImageButton.propTypes = {
  onPress: PropTypes.func,
  style: PropTypes.any,
  imageStyle: PropTypes.any
}

export default ImageButton
