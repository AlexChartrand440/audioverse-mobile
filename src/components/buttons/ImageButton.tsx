import React from 'react'
import {
  TouchableOpacity,
  Image,
  ImageSourcePropType,
} from 'react-native'

interface Props {
  onPress: () => void
  style?: {}
  imageStyle?: {}
  accessibilityLabel?: string,
  source: ImageSourcePropType
}

export const ImageButton: React.FC<Props> = ({
  onPress,
  style,
  imageStyle,
  accessibilityLabel,
  source,
  ...props
}) => {
    return (
      <TouchableOpacity
        onPress={onPress}
        style={style}
        accessible={true}
        accessibilityLabel={accessibilityLabel}
        accessibilityRole="button">
        <Image source={source} {...props} style={imageStyle} />
      </TouchableOpacity>
    )
}

export default ImageButton
