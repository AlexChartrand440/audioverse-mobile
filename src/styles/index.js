import { StyleSheet } from 'react-native'

export const GlobalStyles = StyleSheet.create({
  header: {
    backgroundColor: '#E53935',
    elevation: 0,
    borderBottomWidth: 0
  },
  tab: {
    backgroundColor: '#E53935'
  },
  tabIndicator: {
    backgroundColor: '#FFFFFF'
  }
})

export const headerTintColor = '#FFFFFF'
