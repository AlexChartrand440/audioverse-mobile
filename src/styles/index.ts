import { StyleSheet } from 'react-native'
import image from '../../assets/default.png'

export const primaryColor = '#E53935'

export const GlobalStyles = StyleSheet.create({
  header: {
    backgroundColor: primaryColor,
    elevation: 0,
    borderBottomWidth: 0,
    shadowOpacity: 0,
  },
  tab: {
    backgroundColor: primaryColor
  },
  tabIndicator: {
    backgroundColor: '#FFFFFF'
  }
})

export const headerTintColor = '#FFFFFF'
export const defaultImage = image

const htmlTextStyle = {
  fontSize: 18,
  textAlign: 'center',
}
export const HTMLStyles =  {
  p: {
    ...htmlTextStyle,
    marginBottom: 20,
  },
  div: htmlTextStyle,
  li: {
    fontSize: 20,
  },
}
