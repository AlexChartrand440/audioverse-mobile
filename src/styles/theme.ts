// Derived from https://www.npmjs.com/package/react-native-themed-styles

import { ImageStyle, TextStyle, useColorScheme, ViewStyle } from 'react-native';

type AppearanceProvider<T> = () => T;

type NamedStyles<T> = { [P in keyof T]: ViewStyle | TextStyle | ImageStyle };

interface StyleSheetData<N extends string, T, S> {
	styles: Record<N, S>;
	themes: Record<N, T>;
	appearanceProvider: AppearanceProvider<N>;
}

interface IUseThemeResult<S, T, N> {
	styles: S;
	theme: T;
	resolvedName: N;
}

export function registerThemes<N extends string, T, R extends N>(
	themes: Record<N, T>,
	appearanceProvider: AppearanceProvider<R>
) {
	return <S extends NamedStyles<S> | NamedStyles<any>>(fn: (theme: T) => S): StyleSheetData<N, T, S> => {
		const styles: any = {};
		for (const [name, theme] of Object.entries(themes)) {
			styles[name] = fn(theme as T);
		}
		return { styles, themes, appearanceProvider };
	};
}

export function useTheme<T, N extends string, S extends NamedStyles<S> | NamedStyles<any>>(
	data: StyleSheetData<N, T, S>,
	name?: N
): IUseThemeResult<S, T, N> {
	const resolvedName = name || data.appearanceProvider();
	const theme = data.themes[resolvedName];
	if (!theme) {
		throw new Error(`Theme not defined: ${resolvedName}`);
	}
	const styles = data.styles[resolvedName];

	return { styles, theme, resolvedName };
}

export const globalColors = {
	brandPrimary: '#BC2102',
	brandPrimaryDark: '#B01F02', // -5 brightness from primary color
	buttonPrimary: '#D73352',
	black: '#000000',
	white: '#ffffff',
	grey100: '#F4F4F4',
	grey200: '#E9E9EF',
	grey250: '#E0E0E0',
	grey300: '',
	grey400: '#DDDDDD',
	grey500: '#CCCCCC',
	grey600: '#888888',
	grey617: '#9E9E9E',
	grey700: '#666666',
	grey800: '#242424',
	grey850: '#212121',
	grey875: '#1A1917',
	grey900: '#181818',
	grey950: '#121212',
	linkColor: '#337AB7',
	facebookBlue: '#4E64B2',
} as const;

export const styleSheetFactory = registerThemes(
	{
		light: {
			...globalColors,
			isDark: false,
			backgroundColor: 'white',
			textColor: 'black',
			primary: globalColors.brandPrimary,
		},
		dark: {
			...globalColors,
			isDark: true,
			backgroundColor: 'black',
			textColor: 'white',
			primary: globalColors.brandPrimaryDark,
		},
	},
	() => {
		const colorScheme = useColorScheme();
		console.log('styles', colorScheme);
		return colorScheme === 'dark' ? 'dark' : 'light';
	}
);
