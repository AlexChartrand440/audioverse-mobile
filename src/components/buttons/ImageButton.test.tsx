import React from 'react'
import ImageButton from './ImageButton'
import { defaultImage } from '../../styles'

import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const tree = renderer.create(<ImageButton onPress={() => {}} source={defaultImage} />).toJSON()
  expect(tree).toMatchSnapshot()
})
