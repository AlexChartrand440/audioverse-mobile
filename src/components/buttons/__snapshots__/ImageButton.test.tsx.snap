// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Image
    source={1}
  />
</View>
`;
